cmd/root.go

```diff
@@ -53,14 +63,50 @@ func New() *cobra.Command {
 	})
 
 	flags := rootCmd.PersistentFlags()
-	flags.String("config", "", "Config file (default is $HOME/.hubble.yaml)")
-	flags.BoolP("debug", "D", false, "Enable debug messages")
-	flags.String("server", defaults.GetDefaultSocketPath(), "Address of a Hubble server")
-	flags.Duration("timeout", defaults.DefaultDialTimeout, "Hubble server dialing timeout")
+	flags.String("config", "", "Config file (default is $HOME/.hubble.yaml).")
+	flags.BoolP("debug", "D", false, "Enable debug messages.")
+	flags.String("server", defaults.GetDefaultSocketPath(), "Address of a Hubble server.")
+	flags.Duration("timeout", defaults.DefaultDialTimeout, "Hubble server dialing timeout.")
+	flags.Bool(
+		"tls",
```

Comment:
nit: Maybe we want to name this `--tls-required` and (as mentioned above) implicitly enable TLS whenever one of the following applies:

- The scheme starts with `tls://`
- The user has provided a client certificate
- (maybe?) The user has provided any of the other TLS flags (servername, ca)

The semantics of the `tls-required` would be:
- `false`: Enable TLS if one of the above applies
- `true`: Enable TLS unconditionally